r2/r2/lib/media.py

```diff
@@ -205,6 +452,83 @@ def optimize_jpeg(filename):
     with open(os.path.devnull, 'w') as devnull:
         subprocess.check_call(("/usr/bin/jpegoptim", filename), stdout=devnull)
 
+class _SVGSoup(BeautifulSoup.BeautifulStoneSoup):
+    # Fixes unexpected conversion from `<g><g></g></g>` into `<g></g><g></g>`
+    nests = ('defs', 'g', 'marker', 'mask', 'pattern', 'svg', 'symbol', 'use')
+    NESTABLE_TAGS = BeautifulSoup.buildTagMap([], nests)
+
+def _sanitize_svg(image):
+    elements = [x.lower() for x in _SAFE_SVG_ELEMENTS]
+    attributes = [x.lower() for x in _SAFE_SVG_ATTRIBUTES]
```

Comment:
These var names are kind of misleading, maybe `element_whitelist` and `attribute_whitelist`?